salt/_modules/metalk8s_volumes.py

```diff
@@ -339,6 +378,27 @@ def clean_up(self):
             log.warning('{} already removed'.format(exn.filename))
 
 
+class SparseLoopDeviceNoFormat(SparseLoopDevice):
+    # Recent losetup support `--nooverlap` but not the one shipped with
+    # CentOS 7.
+    PROVISIONING_COMMAND = ('losetup', '--find', '--partscan')
+
+    @property
+    def persistent_path(self):
+        return '/dev/disk/by-partlabel/{}'.format(self.get('metadata.uid'))
+
+    @property
+    def is_prepared(self):
+        """Check if the volume is already prepared."""
+        # Partition 1 should be labelled with the volume UUID.
+        pattern = '1.+{}'.format(self.get('metadata.uid').lower())
+        result  = _run_cmd(' '.join(['parted', self.path, 'print']))
```

Comment:
This requires `parted` to be installed. Can we build this on top of plain `sfdisk` instead?

Reply:
From `man 8 sfdisk`:

> sfdisk doesn't understand the GUID Partition Table (GPT) format and it is not designed for large partitions.  In these cases use the more advanced GNU parted(8).

So I guess `parted` is a safer choice (will need to install it through Salt to be sure it's present though).

Reply:
Modern `fdisk` does know GPT, but maybe not on CentOS 7. Anyway, on those systems there should be `sgdisk` for `fdisk`-like GPT management.

Reply:
```
[root@test-bootstrap centos]# sgdisk
bash: sgdisk: command not found
```

Seems not installed by default, so what's the advantage of using it instead of `parted` (since I'll have to make sure it's installed through Salt anyway)?

Reply:
Fair enough. I was thinking `parted` has (many) more deps than `gdisk`, but that seems not to be the case (they're comparable), so if `parted` does what you need, go for it.

Reply:
Done, I'm now using `sgdisk` since it's possible to set the partition GUID with it.